### PR TITLE
overlord/snapstate: remove Download from backend

### PR DESCRIPTION
**Pull request by chipaca**

### CODE REVIEW

**Comment by pedronis** [comment]
would not have been quicker and it would also have preserved more the checking of order to make the fakeBackend a fakeStore too? (it already has code to fake readSnapInfo for example anyway)

**Comment by chipaca** [comment]
yes, probably. But also made the tests slightly harder to follow I think (and they already aren't that easy to follow)

File: overlord/snapstate/backend.go [hunk 1]
```diff
@@ -28,8 +28,8 @@ import (
 
 // A StoreService can find, list available updates and offer for download snaps.
```
Comment (pedronis) [inline overlord/snapstate/backend.go]: drive-by: offers to?

File: overlord/snapstate/snapmgr_test.go [hunk 2]
```diff
@@ -753,20 +752,11 @@ func (s *snapmgrTestSuite) TestUpdateSameRevisionRunThrough(c *C) {
...
 
-	expected := []fakeOp{
-		{
-			op:       "download",
-			macaroon: s.user.Macaroon,
-			name:     "some-snap",
-			channel:  "channel-for-7",
-		},
-	}
-
```
Comment (pedronis) [inline overlord/snapstate/snapmgr_test.go]: maybe is just the diff that is hard to read but does this have a matching "s.fakeStore.downloads, DeepEquals" ?

**Comment by pedronis** [comment]
wondering if it would be worth to chain the fakeStore to the fakeBackend to get a just a "download" fakeOp in the list there as well? overkill?

File: overlord/snapstate/snapmgr_test.go (hunk 2 above)
Comment (chipaca) [inline overlord/snapstate/snapmgr_test.go]: whoops, missed this one

File: overlord/snapstate/backend.go (hunk 1 above)
Comment (chipaca) [inline overlord/snapstate/backend.go]: will do

**Comment by pedronis** [comment]
LGTM, thank you

**Comment by mvo5** [comment]
:+1:
